generated/niscope/session.py

```diff
@@ -1839,6 +1839,127 @@ def configure_vertical(self, range, coupling, offset=0.0, probe_attenuation=1.0,
+
+                                    -  **relative_initial_x** the time (in seconds) from the trigger to the first sample in the fetched waveform
+                                    -  **absolute_initial_x** timestamp (in seconds) of the first fetched sample. This timestamp is comparable between records and acquisitions; devices that do not support this parameter use 0 for this output.
+                                    -  **x_increment** the time between points in the acquired waveform in seconds -  **actual_samples** the actual number of samples fetched and placed in the waveform array
+                                    -  **gain** the gain factor of the given channel; useful for scaling binary data with the following formula:
+
+                                        .. math::
+
+                                            voltage = binary data * gain factor + offset
+
+                                    -  **offset** the offset factor of the given channel; useful for scaling binary data with the following formula:
+
+                                        .. math::
+
+                                            voltage = binary data * gain factor + offset
+
+                                    Call _actual_num_wfms to determine the size of this array.
+
+        '''
+        import sys
+
+        if num_samples is None and wfm is None:
+            raise TypeError("Either 'num_samples' or 'wfm' must be set")
+
+        # Check optional parameters
+        if fetch_relative_to is not None:
+            self.fetch_relative_to = fetch_relative_to
+        if fetch_offet is not None:
+            self.fetch_offet = fetch_offet
+        if fetch_record_number is not None:
+            self.fetch_record_number = fetch_record_number
+        if fetch_num_records is not None:
+            self.fetch_num_records = fetch_num_records
+
+        num_wfms = self._actual_num_wfms()
+
+        if num_samples is None:
+            num_samples_to_use = int(len(wfm) / num_wfms)
+        else:
+            num_samples_to_use = num_samples
+
+        total_wfm_size = num_samples_to_use * num_wfms
+
+        # If the waveform is not passed in, we will use the builtin python array with element type 'double'
+        if wfm is None:
+            wfm_to_use, wfm_info = self._fetch(num_samples, timeout)
+        else:
+            if len(wfm) < total_wfm_size:
+                raise TypeError('The size of the input array is too small for the acquisition. Needs to be {0}, was {1}'.format(total_wfm_size, len(wfm_to_use)))
+
+            wfm_to_use = wfm
+            wfm_info = self.fetch_into(wfm_to_use, timeout)
+
+        # memoryview in Python 2 doesn't support numeric types, so we copy into an array.array to put in the wfm. :( You should be using Python 3!
+        # Or use the _into version. memoryview in Python 2 only supports string and bytearray, not array.array or numpy.ndarray of arbitrary types.
+        if sys.version_info.major < 3:
```

Comment:
can six help in any way for this?

Reply:
I don't see anything in `six` for `memoryview`'s